src/configure/services/gitHubService.ts

```diff
@@ -0,0 +1,29 @@
+export class GitHubProvider {
+    // private gitHubPatToken: string;
+    private static GitHubUrl = "https://github.com/";
+
+    // constructor(gitHubPat: string) {
+    //     this.gitHubPatToken = gitHubPat;
+    // }
+
+    public static isGitHubUrl(remoteUrl: string): boolean {
+        return remoteUrl.indexOf(GitHubProvider.GitHubUrl) >= 0;
+    }
+
+    public static getRepositoryIdFromUrl(remoteUrl: string): string {
+        let endCount: number = remoteUrl.indexOf(".git");
+        if (endCount <= 0) {
```

Comment:
what is this scenario ?

Reply:
This is used while sourceRepositoryProvide is forming GitRepositoryDetails object with local git details.
As the provider need  not know how to extract repoId or repoName from a GitHub url or Azure Repo's URL, hence exposed this logic as a function in the respective services.

Reply:
I meant i didn't understand the scenario for endcount<=0 ? Should the url always start with GitHubProvider.GitHubUrl?

Reply:
As discussed offline, will remove the =0 condition to remove confusion

Reply:
And checking startsWith in isGithubUrl